python/ray/data/tests/test_dataset_formats.py

```diff
@@ -932,15 +932,26 @@ def test_parquet_read_parallel_meta_fetch(ray_start_regular_shared, fs, data_pat
 
 
 def test_parquet_reader_estimate_data_size(shutdown_only, tmp_path):
-    ds = ray.data.range(1000)
-    path = os.path.join(tmp_path, "test_parquet_dir")
-    os.mkdir(path)
-    ds.repartition(30).write_parquet(path)
+    tensor_output_path = os.path.join(tmp_path, "tensor")
+    ray.data.range_tensor(1000, shape=(1000,)).write_parquet(tensor_output_path)
+    reader = _ParquetDatasourceReader(tensor_output_path)
+    assert (
+        reader._encoding_ratio >= 400 and reader._encoding_ratio <= 600
+    ), "encoding ratio is out of expected bound"
+    data_size = reader.estimate_inmemory_data_size()
+    assert (
+        data_size >= 7_000_000 and data_size <= 10_000_000
```

Comment:
This looks quite accurate! Have you tried some real data files?

Reply:
@jianoaix - I also tried the mock Uber dataset in nightly test. For `s3://shuffling-data-loader-benchmarks/data/r10_000_000_000-f1000/input_data_999.parquet.snappy`, this framework estimates each row data size to be 198 bytes. The input table has 23 int64/float columns. So the estimation size is quite close.

Reply:
Nice, note those are not real data. Let me point the user to this PR and ask them to give it a try.

Reply:
@c21 We should try this with the NYC taxi dataset, it's hosted in a public Ursa Labs bucket and is used in this example: https://docs.ray.io/en/master/data/examples/nyc_taxi_basic_processing.html#reading-and-inspecting-the-data

Reply:
@clarkzinzow, @ericl and @jianoaix - here is the newly estimated size after this PR for the example in https://docs.ray.io/en/master/data/examples/nyc_taxi_basic_processing.html :

```
>>> ds.size_bytes()
2461596040
>>> ds.fully_executed().size_bytes()
2263031675
```

This is more accurate now (and good thing is it has some overestimation now), I will update the doc in a followup PR.

Reply:
Wow, this is super close!